### ISSUE TITLE
mkosi.postinst etc/resolv.conf is not world-readable

### ISSUE THREAD

**Issue by lights0123** [systemd/mkosi#2239]
Trying to install an Ubuntu Focal image with mkosi v19, `apt update` in my mkosi.postinst.chroot is failing with a network error. This is because the sandbox Apt user, `_apt`, does not have read access to `/run/systemd`. To debug this, I ran the following script as mkosi.postinst.chroot:

```
ls -l /etc/resolv.conf
ls -l /run/systemd/resolve/stub-resolv.conf
cat /etc/resolv.conf
su -s /bin/bash _apt -c 'ls -l /etc/resolv.conf'
su -s /bin/bash _apt -c 'ls -l /run'
su -s /bin/bash _apt -c 'ls -l /run/systemd'
su -s /bin/bash _apt -c 'ls -l /run/systemd/resolve'
su -s /bin/bash _apt -c 'ls -l /run/systemd/resolve/stub-resolv.conf'
su -s /bin/bash _apt -c 'cat /etc/resolv.conf'
```

Its output was:

```
+/work/postinst:6> ls -l /etc/resolv.conf
lrwxrwxrwx 1 root root 39 Jan  3 21:00 /etc/resolv.conf -> ../run/systemd/resolve/stub-resolv.conf
+/work/postinst:7> ls -l /run/systemd/resolve/stub-resolv.conf
-rw-r--r-- 1 root root 55 Jan  3 19:38 /run/systemd/resolve/stub-resolv.conf
+/work/postinst:8> cat /etc/resolv.conf
# Generated by NetworkManager
nameserver 192.168.100.1
+/work/postinst:9> su -s /bin/bash _apt -c 'ls -l /etc/resolv.conf'
lrwxrwxrwx 1 root root 39 Jan  3 21:00 /etc/resolv.conf -> ../run/systemd/resolve/stub-resolv.conf
+/work/postinst:10> su -s /bin/bash _apt -c 'ls -l /run'
total 0
drwxr-xr-x 2 root root 60 Jan  3 21:08 host
drwx------ 3 root root 60 Jan  3 21:08 systemd
+/work/postinst:11> su -s /bin/bash _apt -c 'ls -l /run/systemd'
ls: cannot open directory '/run/systemd': Permission denied
‣ "mkosi-chroot /work/postinst final" returned non-zero exit code 2.
```

So, it appears that `/run/systemd` needs to have world-read permissions.

**Comment by DaanDeMeyer** [comment]
Note that you don't have to chroot to install packages. You can just run `apt` in `mkosi.postinst` and it'll automatically operate on the image instead of on the host. We should still fix this however. It's caused by bwrap using rather restrictive permissions when creating directories. We should probably make it less restrictive. Problem is the options to make it less restrictive were only introduced in bwrap 0.5.0 which isn't available in e.g. CentOS 8 Stream.

**Comment by DaanDeMeyer** [comment]
This should be fixed by #2255. Please let me know if it's still not working
